odoo/fields.py

```diff
@@ -1054,20 +1056,20 @@ def update_db_notnull(self, model, column):
         """
         has_notnull = column and column['is_nullable'] == 'NO'
 
-        if not column or (self.required and not has_notnull):
+        if not column or (self.required is True and not has_notnull):
```

Comment:
You already check that it is a boolean, then don't modify that everywhere (`is True` is not pythonic and ugly IMO)?